nix/default.nix

```diff
@@ -1,45 +1,52 @@
 { system ? builtins.currentSystem }:
 let
-  pkgs = import ./nixpkgs.nix {};
+  pkgs = (import ./nixpkgs.nix {
+    config = {
+      packageOverrides = pkg: {
+        gpgme = (static pkg.gpgme);
+        libassuan = (static pkg.libassuan);
+        libgpgerror = (static pkg.libgpgerror);
+        libseccomp = (static pkg.libseccomp);
+        systemd = pkg.systemd.overrideAttrs(x: {
+          mesonFlags = x.mesonFlags ++ [ "-Dstatic-libsystemd=true" ];
+          postFixup = ''
+            ${x.postFixup}
+            sed -ri "s;$out/(.*);$nukedRef/\1;g" $lib/lib/libsystemd.a
+          '';
+        });
+      };
+    };
+  });
 
-  static = pkg: pkg.overrideAttrs(old: {
-    configureFlags = (old.configureFlags or []) ++
+  static = pkg: pkg.overrideAttrs(x: {
+    configureFlags = (x.configureFlags or []) ++
       [ "--without-shared" "--disable-shared" ];
     dontDisableStatic = true;
     enableSharedExecutables = false;
     enableStatic = true;
   });
 
-  patchLvm2 = pkg: pkg.overrideAttrs(old: {
-    configureFlags = [
-      "--disable-cmdlib" "--disable-readline" "--disable-udev_rules"
-      "--disable-udev_sync" "--enable-pkgconfig" "--enable-static_link"
-    ];
-    preConfigure = old.preConfigure + ''
-      substituteInPlace libdm/Makefile.in --replace \
-        SUBDIRS=dm-tools SUBDIRS=
-      substituteInPlace tools/Makefile.in --replace \
-        "TARGETS += lvm.static" ""
-      substituteInPlace tools/Makefile.in --replace \
-        "INSTALL_LVM_TARGETS += install_tools_static" ""
-    '';
-    postInstall = "";
-  });
-
-  self = {
-    cri-o-static = (pkgs.cri-o.overrideAttrs(old: {
+  self = with pkgs; {
+    cri-o-static = (cri-o.overrideAttrs(x: {
       name = "cri-o-static";
       src = ./..;
-      BUILDTAGS="${old.BUILDTAGS} netgo";
-      EXTRA_LDFLAGS = ''-linkmode external -extldflags "-static -lm"'';
-      dontStrip = true;
-      # DEBUG = 1; # Uncomment this line to enable debug symbols in the binary
-    })).override {
-      gpgme = (static pkgs.gpgme);
-      libassuan = (static pkgs.libassuan);
-      libgpgerror = (static pkgs.libgpgerror);
-      libseccomp = (static pkgs.libseccomp);
-      lvm2 = (patchLvm2 (static pkgs.lvm2));
-    };
+      doCheck = false;
+      buildInputs = [
+        glibc
+        glibc.static
+        gpgme
+        libapparmor
+        libassuan
+        libgpgerror
+        libseccomp
+        libselinux
+        systemd
```

Comment:
Systemd should not be needed to build CRI-O

Reply:
Thank you for clarify ;-)